compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1980,46 +2005,67 @@ setTransposeLikeOpRootConfig(mlir::FunctionOpInterface entryPointFn,
   assert(!getLoweringConfig(genericOp) &&
          "expected lowering_config is not set");
 
+  if (!linalgOpInfo.isTranspose())
+    return failure();
+
   LLVM_DEBUG(KD_DBGS() << "Setting transpose-like op root configuration\n");
 
+  DistributionHeuristicConfig distConfig;
   auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(entryPointFn);
-  if (!hasAVX2Feature(targetAttr) ||
-      !x86TransposeLoweringPrecondition(genericOp)) {
-    return failure();
-  }
+  bool targetIsAArch64SME = isAArch64(targetAttr) &&
+                            hasSMEFeature(targetAttr) &&
+                            clEnableScalableVectorization;
+  if (hasAVX2Feature(targetAttr)) {
```

Comment:
The big if-else makes readability really bad. Can we create methods to get `minTileSizes`? So it can be like `std::optional<SmallVector<int64_t>> getX86TransposeVectorSizes()` and `std::optional<SmallVector<int64_t>> getAArch64TransposeVectorSizes()`. What do you think?